File: BookShop_backend/src/main/java/com/example/computer_shoponline/service/TaiKhoanService.java
package com.example.computer_shoponline.service;

import com.example.computer_shoponline.dto.TaiKhoan_Dto;

import java.util.List;

public interface TaiKhoanService {
    TaiKhoan_Dto getTaiKhoanByID(int taiKhoanId);
    List<TaiKhoan_Dto> getAllTaiKhoan();
    TaiKhoan_Dto createTaiKhoan(TaiKhoan_Dto taiKhoan);
    TaiKhoan_Dto updateTaiKhoan(int taiKhoanId, TaiKhoan_Dto taiKhoan);
    void deleteTaiKhoan(int taiKhoanId);
}

File: BookShop-forntend/src/service/TaiKhoanService.js
import axios from 'axios'

const REST_API_BASE_URL = 'http://localhost:9090/api/taikhoans'


export const getAllTaiKhoan = () => axios.get(REST_API_BASE_URL);

export const createTaikhoan = (taikhoans) => axios.post(REST_API_BASE_URL, taikhoans);
export const getTaiKhoanById = (taikhoanid) => axios.get(REST_API_BASE_URL + '/' + taikhoanid)
export const updateTaikhoan = (taikhoanid, taikhoans) => axios.put(REST_API_BASE_URL + '/' + taikhoanid, taikhoans)
export const deleteTaikhoan = (taikhoanid) => axios.delete(REST_API_BASE_URL + '/' + taikhoanid)



